resolwe_bio/processes/import_data/sra_file.py

```diff
@@ -0,0 +1,399 @@
+"""Import-SRA."""
+import glob
+from pathlib import Path
+
+from plumbum import TEE
+from shutil import move
+from resolwe.process import (
+    BooleanField,
+    Cmd,
+    FileField,
+    FileHtmlField,
+    GroupField,
+    IntegerField,
+    ListField,
+    Persistence,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+def run_fastqc(fastqs, output_dir):
+    """Run fastQC on given FASTQs.
+
+    :param list fastqs: List of fastqs
+    :param str output_dir: Output directory
+
+    """
+    output_path = Path(output_dir)
+    output_path.mkdir(exist_ok=True)
+
+    cmd = Cmd["fastqc"]
+    for fastq in fastqs:
+        cmd = cmd[fastq]
+    cmd = cmd["--extract"]
+    cmd = cmd[f"--outdir={str(output_path)}"]
+    _, _, stderr = cmd & TEE
+
+    return stderr
+
+
+class ImportSra(Process):
+    """Import reads from SRA.
+
+    Import single or paired-end reads from Sequence Read Archive (SRA)
+    via an SRA accession number. SRA stores raw sequencing data and
+    alignment information from high-throughput sequencing platforms.
+
+    """
+    slug = "import-sra"
+    name = "SRA data"
+    process_type = "data:sra"
+    version = "1.0.0"
+    category = "Import"
+    scheduling_class = SchedulingClass.BATCH
+    persistence = Persistence.TEMP
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {
+                "image": "resolwebio/common:1.3.1"
+            }
+        },
+        "resources": {
+            "cores": 1,
+            "memory": 1024,
+            "network": True,
+        },
+    }
+    data_name = "{{ sra_accession|first }}"
+
+    class Input:
+        """Input fields to process ImportSra."""
+        sra_accession = ListField(StringField(), label="SRA accession(s)")
+        show_advanced = BooleanField(label="Show advanced options", default=False)
+
+        class Advanced:
+            """Advanced options."""
+            min_spot_id = IntegerField(label="Minimum spot ID", required=False)
+            max_spot_id = IntegerField(label="Maximum spot ID", required=False)
+            min_read_len = IntegerField(label="Minimum read length", required=False)
+            clip = BooleanField(label="Clip adapter sequences", default=False)
+            aligned = BooleanField(label="Dump only aligned sequences", default=False)
+            unaligned = BooleanField(label="Dump only unaligned sequences", default=False)
+
+        advanced = GroupField(Advanced, label="Advanced options", hidden="!show_advanced")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+        first_read = Cmd["fastq-dump"]("-X", 1, "-Z", "--split-spot", inputs.sra_accession[0])
```

Comment:
This test could be more thorough, failing for SRAs of mixed types.

Reply:
Agreed, I will add that.